### PR TITLE
Adding ability to set tags for network firewall rules

### PR DESCRIPTION
**Pull request by endzyme**
This is needed for private clusters to allow master private IPv4 CIDR to talk to admission webhooks that live in the cluster. This is done via tags and firewall rules. Adding tags to an existing node-pool _will_ recreate it so be advised!

### CODE REVIEW

**Comment by endzyme** [comment]
So **big notes** here:
1. If you were to change from `node-pool-v1.0.1` to `node-pool-v2.0.0` you would have to recreate all your node pools :cry: - I should probably note that in the changelog.
2. There isn't a way to just add the tags feature since "create-before-destroy" exists for the node pool. Which means there is a name conflict when you change that value
3. Any other considerations we should make for changes that have this type of impact?

**Comment by endzyme** [comment]
I've merged in #10 so that we limit the amount of `keepers` we change between versions. Any keepers changes always result in a major version bump due to the recreate issue.

**Comment by endzyme** [comment]
@emailbob (Thanks!!)

File: node_pool/main.tf [hunk 1]
```diff
@@ -18,6 +18,9 @@ resource "random_id" "entropy" {
     name         = "${var.name}"
     region       = "${var.region}"
     disk_size    = "${var.disk_size_in_gb}"
+    tags         = "${join(",", sort(var.node_pool_tags))}"
+    disk_type    = "${var.disk_type}"
+    labels       = "${jsonencode(var.node_labels)}"
```
Comment (endzyme) [inline node_pool/main.tf]: After testing - @emailbob verified that this `jsonencode()` function will sort by the keys and keep consistent orders if the order changes.

**Comment by endzyme** [comment]
I know lots of people are in and out over the next few days but any time to help review would be awesome :)

**Comment by bambash** [comment]
:+1: provisioned a cluster this morning with the new commits, everything works as expected

**Comment by emailbob** [comment]
One more addition to this PR. I have a client who wants the nodes pool to start off at zero. Currently the initial_node_count is hardcoded at 1. I updated this so the default is still 1 but now we can override this.

Warning since changing the initial_node_count value will recreate the resource I had to add it to the keepers which will also cause the resource to be recreated.

File: node_pool_taint/example-usage [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# In the below example, `module.customername_cluster` is an instance of
+# one of the GKE modules from this repository.
+
+module "customername_cluster_node_pool" {
+  source = "git@github.com:/FairwindsOps/terraform-gke//node_pool_taint"
```
[the remark this answers is not in the document]
Comment (emailbob) [reply node_pool_taint/example-usage]: @dosullivan Thanks for the feedback. Those updates have been added.

**Pull request merged**